urla134d66615_fikr4n_itl_java_tgz-pJ8-example_ExampleJ8/src/org/arabeyes/itl/prayer/astro/AstroLib.java

```diff
@@ -0,0 +1,491 @@
+		{ 11, 0, 0, 0 }, { -10, 0, 5, 0 }, { -8, 0, 3, 0 },
+		{ 7, 0, -3, 0 }, { -7, 0, 0, 0 }, { -7, 0, 3, 0 }, { -7, 0, 3, 0 },
+		{ 6, 0, 0, 0 }, { 6, 0, -3, 0 }, { 6, 0, -3, 0 }, { -6, 0, 3, 0 },
+		{ -6, 0, 3, 0 }, { 5, 0, 0, 0 }, { -5, 0, 3, 0 }, { -5, 0, 3, 0 },
+		{ -5, 0, 3, 0 }, { 4, 0, 0, 0 }, { 4, 0, 0, 0 }, { 4, 0, 0, 0 },
+		{ -4, 0, 0, 0 }, { -4, 0, 0, 0 }, { -4, 0, 0, 0 }, { 3, 0, 0, 0 },
+		{ -3, 0, 0, 0 }, { -3, 0, 0, 0 }, { -3, 0, 0, 0 }, { -3, 0, 0, 0 },
+		{ -3, 0, 0, 0 }, { -3, 0, 0, 0 }, { -3, 0, 0, 0 } };
+	
+	static final int SINCOEFF[][] = { { 0, 0, 0, 0, 1 }, { -2, 0, 0, 2, 2 },
+		{ 0, 0, 0, 2, 2 }, { 0, 0, 0, 0, 2 }, { 0, 1, 0, 0, 0 },
+		{ 0, 0, 1, 0, 0 }, { -2, 1, 0, 2, 2 }, { 0, 0, 0, 2, 1 },
+		{ 0, 0, 1, 2, 2 }, { -2, -1, 0, 2, 2 }, { -2, 0, 1, 0, 0 },
+		{ -2, 0, 0, 2, 1 }, { 0, 0, -1, 2, 2 }, { 2, 0, 0, 0, 0 },
+		{ 0, 0, 1, 0, 1 }, { 2, 0, -1, 2, 2 }, { 0, 0, -1, 0, 1 },
+		{ 0, 0, 1, 2, 1 }, { -2, 0, 2, 0, 0 }, { 0, 0, -2, 2, 1 },
+		{ 2, 0, 0, 2, 2 }, { 0, 0, 2, 2, 2 }, { 0, 0, 2, 0, 0 },
+		{ -2, 0, 1, 2, 2 }, { 0, 0, 0, 2, 0 }, { -2, 0, 0, 2, 0 },
+		{ 0, 0, -1, 2, 1 }, { 0, 2, 0, 0, 0 }, { 2, 0, -1, 0, 1 },
+		{ -2, 2, 0, 2, 2 }, { 0, 1, 0, 0, 1 }, { -2, 0, 1, 0, 1 },
+		{ 0, -1, 0, 0, 1 }, { 0, 0, 2, -2, 0 }, { 2, 0, -1, 2, 1 },
+		{ 2, 0, 1, 2, 2 }, { 0, 1, 0, 2, 2 }, { -2, 1, 1, 0, 0 },
+		{ 0, -1, 0, 2, 2 }, { 2, 0, 0, 2, 1 }, { 2, 0, 1, 0, 0 },
+		{ -2, 0, 2, 2, 2 }, { -2, 0, 1, 2, 1 }, { 2, 0, -2, 0, 1 },
+		{ 2, 0, 0, 0, 1 }, { 0, -1, 1, 0, 0 }, { -2, -1, 0, 2, 1 },
+		{ -2, 0, 0, 0, 1 }, { 0, 0, 2, 2, 1 }, { -2, 0, 2, 0, 1 },
+		{ -2, 1, 0, 2, 1 }, { 0, 0, 1, -2, 0 }, { -1, 0, 1, 0, 0 },
+		{ -2, 1, 0, 0, 0 }, { 1, 0, 0, 0, 0 }, { 0, 0, 1, 2, 0 },
+		{ 0, 0, -2, 2, 2 }, { -1, -1, 1, 0, 0 }, { 0, 1, 1, 0, 0 },
+		{ 0, -1, 1, 2, 2 }, { 2, -1, -1, 2, 2 }, { 0, 0, 3, 2, 2 },
+		{ 2, -1, 0, 2, 2 } };
+	
+	public static double getRefraction(final Location loc, double sunAlt) {
+		double part1, part2;
+		
+		part1 = (loc.getPressure() / 1010.0)
+		* (283 / (273 + loc.getTemperature()));
+		part2 = 1.02 / (Utils.radToDeg(Math.tan(Utils.degToRad(sunAlt + (10.3/(sunAlt + 5.11))))) + 0.0019279);
+		return (part1 * part2) / 60.0;
+	}
+	
+	public static double getJulianDay(final SimpleGregorianDate date, double gmt) {
+		
+		double jdB = 0, jdY, jdM, JD;
+		
+		jdY = date.getYear();
+		jdM = date.getMonthCompat();
+		
+		if (date.getMonthCompat() <= 2) {
+			jdY--;
+			jdM += 12;
+		}
+		
+		if (date.getYear() < 1)
+			jdY++;
+		
+		if ((date.getYear() > 1582)
+				|| ((date.getYear() == 1582) && ((date.getMonthCompat() > 10) || ((date
+						.getMonthCompat() == 10) && (date.getDay() >= 4)))))
+			jdB = 2 - Math.floor(jdY / 100.0) + Math.floor((jdY / 100.0) / 4.0);
+		
+		JD = Math.floor(365.25 * (jdY + 4716.0))
+		+ Math.floor(30.6001 * (jdM + 1)) + (date.getDay() + (-gmt) / 24.0)
+		+ jdB - 1524.5;
+		
+		return JD;
+	}
+	
+	public static void getAstroValuesByDay(double julianDay,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 57 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.